### DISCUSSION TITLE
How to gracefully shutdown GUI software using Command+Q on Mac? (CloseRequested event seems to be ignored)

### DISCUSSION THREAD

**Discussion by TheQuantumPhysicist** [hecrj/iced#1950] [Q&A]
When we use the keyboard shortcut Command+Q on Mac to close the GUI application, it circumvents the `iced::window::Event::CloseRequested` event and just "crashes" the GUI application, in a way, circumventing the proper shutdown procedure. Unlike on Windows, where doing Alt+F4 will do the right thing. Now I'm new to mac so I'm not the expert by a long shot, but I've seen other software close gracefully with Command+Q even though it looks nuclear sometimes. Is there a way in iced to handle this properly? Gurus who use the keyboard for everything won't aim for the "x" button to close the GUI app. They'll just hammer it down with Command+Q. There has to be a handler for this.

**Comment by lufte** [comment]
There is a bug already for this #1941
